osu.Game/Overlays/Chat/Tabs/ChannelTabControl.cs

```diff
@@ -59,16 +59,12 @@ protected override TabItem<Channel> CreateTabItem(Channel value)
 
         /// <summary>
         /// Adds a channel to the ChannelTabControl.
-        /// The first channel added will automaticly selected.
         /// </summary>
         /// <param name="channel">The channel that is going to be added.</param>
         public void AddChannel(Channel channel)
         {
             if (!Items.Contains(channel))
                 AddItem(channel);
-
-            if (Current.Value == null)
```

Comment:
Why was this removed? This will break in the case where you add the first channel (not from the initial load of channels).

Reply:
[I proposed moving this up higher earlier](https://github.com/ppy/osu/pull/7227#discussion_r368304286) to keep this type of logic in one place, but unsure if doing that does break anything.

Reply:
I don't have control which channel is selected without this change. The logic of ChatOverlay is, once loaded, the selected tab gets switched over to the first one, causing an event to fire that switches the ChannelManager's channel. I couldn't await the loading of ChatOverlay (because I'm not well inversed with scheduling inside the framework), which would actually help me to not use this change.

Reply:
The bigger issue here is that `ChannelManager` should already be doing this, but its current channel is listed as `"+"` which should never be the current channel.

Reply:
Have you noticed the changes inside ChatOverlay though?

Reply:
I have, and it is causing the issue mentioned here. Still needs to be fixed.

Reply:
@smoogipoo it might be sleep deprivation talking, but I don't see what the problem is - would you mind elaborating on the problem so I can fix it?

Reply:
I mean, if we would go not modifying ChannelTabControl, we have to rework how ChatOverlay reacts to tab changes, as it's definitely unintentionally causing to change the current channel set from other code (in other words, this one right here).

Reply:
It seems to also be broken on master, but the intention is that the first channel joined is the first channel selected. This happens post-load, e.g. the user can close all channels and then open up a new channel.

Reply:
What I would propose, is that ChatOverlay, if it detects a tab change (from + to *channel) and the current channel (of ChannelManager) is not null, it would set the tab back to the ChannelManager's channel.